File: iconics-java/src/main/java/com/open/iconics/context/IconicsLayoutInflater2.java
package com.open.iconics.context;

import android.content.Context;
import android.util.AttributeSet;
import android.view.View;

import androidx.appcompat.app.AppCompatDelegate;

public class IconicsLayoutInflater2 implements android.view.LayoutInflater.Factory2 {
    private AppCompatDelegate appCompatDelegate;
    private final IconicsFactory mIconicsFactory;

    public IconicsLayoutInflater2(AppCompatDelegate appCompatDelegate) {
        this.appCompatDelegate = appCompatDelegate;
        this.mIconicsFactory = new IconicsFactory();
    }

    @Override
    public View onCreateView(View parent, String name, Context context, AttributeSet attrs) {
        View result = appCompatDelegate.createView(parent, name, context, attrs);
        return mIconicsFactory.onViewCreated(result, context, attrs);
    }

    @Override
    public View onCreateView(String name, Context context, AttributeSet attrs) {
        View result = appCompatDelegate.createView(null, name, context, attrs);
        return mIconicsFactory.onViewCreated(result, context, attrs);
    }
}

File: iconics-kotlin/src/main/java/com/open/iconics/context/IconicsLayoutInflater2.kt
package com.open.iconics.context

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View

import androidx.appcompat.app.AppCompatDelegate

@Deprecated(message = "Use the IconicsImageView or IconicsTextView instead")
class IconicsLayoutInflater2(
    private val appCompatDelegate: AppCompatDelegate
) : LayoutInflater.Factory2 {

    override fun onCreateView(
        parent: View?,
        name: String,
        context: Context,
        attrs: AttributeSet
    ): View? {
        val view = appCompatDelegate.createView(parent, name, context, attrs)
        return IconicsFactory.onViewCreated(view, context, attrs)
    }

    override fun onCreateView(name: String, context: Context, attrs: AttributeSet): View? {
        val view = appCompatDelegate.createView(null, name, context, attrs)
        return IconicsFactory.onViewCreated(view, context, attrs)
    }
}
